paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheet.kt

```diff
@@ -154,13 +164,175 @@ class PaymentSheet internal constructor(
+                allowsDelayedPaymentMethods,
+                appearance
+            )
+        }
+    }
+
+    @Parcelize
+    data class Appearance(
+        // Describes the colors used while the system is in light mode
+        val colorsLight: Colors = Colors.defaultLight,
+
+        // Describes the colors used while the system is in dark mode
+        val colorsDark: Colors = Colors.defaultDark,
+
+        // Describes the appearance of shapes
+        val shapes: Shapes = Shapes.default,
+
+        // Describes the typography used for text.
+        val typography: Typography = Typography.default,
+    ) : Parcelable {
+        fun getColors(isDark: Boolean): Colors {
+            return if (isDark) colorsDark else colorsLight
+        }
+
+        class Builder {
+            private var colorsLight = Colors.defaultLight
+            private var colorsDark = Colors.defaultDark
+            private var shapes = Shapes.default
+            private var typography = Typography.default
+
+            fun colorsLight(colors: Colors) = apply { this.colorsLight = colors }
+            fun colorsDark(colors: Colors) = apply { this.colorsDark = colors }
+            fun shapes(shapes: Shapes) = apply { this.shapes = shapes }
+            fun typography(typography: Typography) = apply { this.typography = typography }
+        }
+    }
+
+    @Parcelize
+    data class Colors(
+        // A primary color used throughout PaymentSheet
+        @ColorRes
+        val primary: Int,
+
+        // The color used for the surfaces (backgrounds) of PaymentSheet
+        @ColorRes
+        val surface: Int,
+
+        // The color used for the background of inputs, tabs, and other components
+        @ColorRes
+        val componentBackground: Int,
+
+        // The color used for borders of inputs, tabs, and other components
+        @ColorRes
+        val componentBorder: Int,
+
+        // The color of the divider lines used inside inputs, tabs, and other components
+        @ColorRes
+        val componentDivider: Int,
+
+        // The default color used for text and on other primary elements in Payment Sheet
+        @ColorRes
+        val onPrimary: Int,
+
+        // The color used for text of secondary importance. For example, this color is used for the label above input fields
+        @ColorRes
+        val textSecondary: Int,
+
+        // The color used for input placeholder text
+        @ColorRes
+        val placeholderText: Int,
+
+        // The color used for items appearing over the background in Payment Sheet
+        @ColorRes
+        val onBackground: Int,
+
+        // / The color used for icons in PaymentSheet, such as the close or back icons
+        @ColorRes
+        val appBarIcon: Int,
+
+        // A color used to indicate errors or destructive actions in PaymentSheet
+        @ColorRes
+        val error: Int,
+    ) : Parcelable {
+        companion object {
+            val defaultLight = Colors(
```

Comment:
Where do we think the best place to define these defaults is? Maybe it makes sense to move these all back to `PaymentsTheme` ? I'm back and forth on this.